Source/Core/DolphinWX/LuaScriptFrame.cpp

```diff
@@ -0,0 +1,553 @@
+int frameAdvance(lua_State* L)
+{
+  u64 currentFrame = Movie::GetCurrentFrame();
+  Core::DoFrameStep();
+
+  // Block until a frame has actually processed
+  // Prevents a script from executing it's main loop more than once per frame.
+  while (currentFrame == Movie::GetCurrentFrame())
+    ;
+  return 0;
+}
+
+int getFrameCount(lua_State* L)
+{
+  lua_pushinteger(L, Movie::GetCurrentFrame());
+  return 1;
+}
+
+int getMovieLength(lua_State* L)
+{
+  if (Movie::IsMovieActive())
+  {
+    lua_pushinteger(L, Movie::GetTotalFrames());
+  }
+  else
+  {
+    luaL_error(L, "No active movie.");
+  }
+  return 1;
+}
+
+int softReset(lua_State* L)
+{
+  ProcessorInterface::ResetButton_Tap();
+  return 0;
+}
+
+int saveState(lua_State* L)
+{
+  int slot = lua_tointeger(L, 1);
+
+  State::Save(slot);
+  return 0;
+}
+
+int loadState(lua_State* L)
+{
+  int slot = lua_tointeger(L, 1);
+
+  State::Load(slot);
+  return 0;
+}
+
+int getAnalog(lua_State* L)
+{
+  lua_pushinteger(L, pad_status->stickX);
+  lua_pushinteger(L, pad_status->stickY);
+
+  return 2;
+}
+
+int setAnalog(lua_State* L)
+{
+  if (lua_gettop(L) != 2)
+  {
+    luaL_error(L, "Incorrect # of parameters passed to setAnalog.\n");
+  }
+
+  u8 xPos = lua_tointeger(L, 1);
+  u8 yPos = lua_tointeger(L, 2);
+
+  pad_status->stickX = xPos;
+  pad_status->stickY = yPos;
+
+  return 0;
+}
+
+// Same thing as setAnalog except it takes polar coordinates
+// Must use an m int the range [0, 128)
+int setAnalogPolar(lua_State* L)
+{
+  int m = lua_tointeger(L, 1);
+  if (m < 0 || m >= 128)
+  {
+    luaL_error(L, "m is outside of acceptable range [0, 128)");
+  }
+
+  // Gotta convert theta to radians
+  double theta = lua_tonumber(L, 2) * M_PI / 180.0;
+
+  // Round to the nearest whole number, then subtract 128 so that our
+  //"origin" is the stick in neutral position.
+  pad_status->stickX = (u8)(floor(m * cos(theta)) + 128);
+  pad_status->stickY = (u8)(floor(m * sin(theta)) + 128);
+
+  return 0;
+}
+
+int getCStick(lua_State* L)
+{
+  lua_pushinteger(L, pad_status->substickX);
+  lua_pushinteger(L, pad_status->substickY);
+
+  return 2;
+}
+
+int setCStick(lua_State* L)
+{
+  pad_status->substickX = lua_tointeger(L, 1);
+  pad_status->substickY = lua_tointeger(L, 2);
+
+  return 0;
+}
+
+int setCStickPolar(lua_State* L)
+{
+  int m = lua_tointeger(L, 1);
+  if (m < 0 || m >= 128)
+  {
+    luaL_error(L, "m is outside of acceptable range [0, 128)");
+  }
+
+  int theta = lua_tointeger(L, 2);
+
+  // Convert theta to radians
+  theta = theta * M_PI / 180.0;
+
+  pad_status->substickX = (u8)(floor(m * cos(theta)) + 128);
+  pad_status->substickY = (u8)(floor(m * sin(theta)) + 128);
+
+  return 0;
+}
+
+int getButtons(lua_State* L)
+{
+  lua_pushinteger(L, pad_status->button);
+  return 1;
+}
+
+// Each button passed in is set to pressed.
+int setButtons(lua_State* L)
+{
+  const char* s = lua_tostring(L, 1);
+
+  for (int i = 0; i < strlen(s); i++)

```

Comment:
`i` should be a `size_t` here (`<cstddef>` needs to be included), since `strlen` returns a `size_t`.